boa/src/vm/code_block.rs

```diff
@@ -529,37 +551,46 @@ impl JsObject {
                     local_env.initialize_binding("arguments", arguments_obj.into(), context)?;
                 }
 
-                // Add argument bindings to the function environment
-                for (i, param) in code.params.iter().enumerate() {
-                    // Rest Parameters
-                    if param.is_rest_param() {
-                        Function::add_rest_param(param, i, args, context, &local_env);
-                        break;
-                    }
+                let arg_count = args.len();
 
-                    let value = match args.get(i).cloned() {
-                        None => JsValue::undefined(),
-                        Some(value) => value,
-                    };
+                // Push function arguments to the stack.
+                let args = match code.params.len().cmp(&args.len()) {
+                    Ordering::Greater => {
+                        let mut v = args.to_vec();
+                        v.extend(vec![JsValue::Undefined; code.params.len() - args.len()]);
+                        v
+                    }
+                    Ordering::Less | Ordering::Equal => args.to_vec(),
+                };
```

Comment:
Same as 721 to 728, `let args = if code.params.len() > args.len() { x } else {y}`